src/cairoWriter/writers/variableDeclarationStatementWriter.ts

```diff
@@ -68,17 +69,17 @@ export class VariableDeclarationStatementWriter extends CairoASTNodeWriter {
         return [writer.write(getDeclarationForId(id))];
       }
     });
+    const solTyp = safeGetNodeType(node.vInitialValue, this.ast.inference);
+    const cairoTyp = CairoType.fromSol(solTyp, this.ast).toString();
+    let namePortion: string;
     if (declarations.length > 1) {
-      return [
-        endent`${documentation}
-          let (${declarations.map((decl) => `mut ${decl}`).join(', ')}) = ${writer.write(
-          node.vInitialValue,
-        )};`,
-      ];
+      namePortion = `(${declarations.map((decl) => `mut ${decl}`).join(', ')})`;
+    } else {
+      namePortion = `mut ${declarations[0]}`;
     }
     return [
       endent`${documentation}
-      let mut ${declarations[0]} = ${writer.write(node.vInitialValue)};`,
+      let ${namePortion}: ${cairoTyp} = ${writer.write(node.vInitialValue)};`,
```

Comment:
If you have more than one declaration then you would only annotate the last one, right? Is that enough?

Reply:
In Cairo several declarations are represented as a tuple. Thus, `cairoTyp` would be a tuple type which gives types to all variables